### PR TITLE
🌱 Bump catalogd to v0.27.0

### PR DESCRIPTION
**Pull request by m1kola**
# Description

Bump catalogd to v0.27.0 to include latest bug fixes & changes.

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -53,9 +53,8 @@ func createTestCatalog(ctx context.Context, name string, imageRef string) (*cata
 			Source: catalogd.CatalogSource{
 				Type: catalogd.SourceTypeImage,
 				Image: &catalogd.ImageSource{
-					Ref:                   imageRef,
-					InsecureSkipTLSVerify: true,
```
Comment (m1kola) [inline test/e2e/e2e_suite_test.go:57]: Need to deal with lack of `InsecureSkipTLSVerify` in E2Es somehow now.

Comment (tmshort) [reply test/e2e/e2e_suite_test.go:57]: @joelanford's recent changes should've handled some of this... He added another CA to be trusted.

Comment (m1kola) [reply test/e2e/e2e_suite_test.go:57]: It did, but on operator-controller side. This is catalogd. https://github.com/operator-framework/catalogd/pull/399 will help, but we still need to modify catalogd deployment on operator-catalog side to be able to use insecure TLS in E2E.

**Comment by m1kola** [comment]
We probably going to skip this version. We need another catalogd release and modify the deployment so that we can use insecure TLS.

**Pull request closed without merging**